src/Microsoft.Data/StringDataFrameColumn.BinaryOperations.cs

```diff
@@ -42,54 +42,74 @@ public override DataFrameColumn Add<T>(T value, bool inPlace = false)
+        public override PrimitiveDataFrameColumn<bool> ElementwiseEquals(DataFrameColumn column)
         {
-            PrimitiveDataFrameColumn<bool> ret = new PrimitiveDataFrameColumn<bool>(Name, Length);
+            return ElementwiseEqualsImplementation(this, column);
+        }
+
+        internal static PrimitiveDataFrameColumn<bool> ElementwiseEqualsImplementation<T>(DataFrameColumn left, T value)
+        {
+            PrimitiveDataFrameColumn<bool> ret = new PrimitiveDataFrameColumn<bool>(left.Name, left.Length);
             string valString = value.ToString();
-            for (long i = 0; i < Length; i++)
+            for (long i = 0; i < left.Length; i++)
             {
-                ret[i] = (string)this[i] == valString;
+                ret[i] = (string)left[i] == valString;
             }
             return ret;
         }
 
-        public override PrimitiveDataFrameColumn<bool> ElementwiseNotEquals(DataFrameColumn column)
+        public override PrimitiveDataFrameColumn<bool> ElementwiseEquals<T>(T value)
         {
-            if (Length != column.Length)
+            return ElementwiseEqualsImplementation(this, value);
+        }
+
+        internal static PrimitiveDataFrameColumn<bool> ElementwiseNotEqualsImplementation(DataFrameColumn left, DataFrameColumn column)
+        {
+            if (left.Length != column.Length)
             {
                 throw new ArgumentException(Strings.MismatchedColumnLengths, nameof(column));
             }
-            PrimitiveDataFrameColumn<bool> ret = new PrimitiveDataFrameColumn<bool>(Name, Length);
-            for (long i = 0; i < Length; i++)
+            PrimitiveDataFrameColumn<bool> ret = new PrimitiveDataFrameColumn<bool>(left.Name, left.Length);
+            for (long i = 0; i < left.Length; i++)
             {
-                ret[i] = (string)this[i] != column[i].ToString();
+                ret[i] = (string)left[i] != column[i].ToString();
             }
             return ret;
         }
 
-        public override PrimitiveDataFrameColumn<bool> ElementwiseNotEquals<T>(T value)
+        public override PrimitiveDataFrameColumn<bool> ElementwiseNotEquals(DataFrameColumn column)
         {
-            PrimitiveDataFrameColumn<bool> ret = new PrimitiveDataFrameColumn<bool>(Name, Length);
+            return ElementwiseNotEqualsImplementation(this, column);
+        }
+
+        internal static PrimitiveDataFrameColumn<bool> ElementwiseNotEqualsImplementation<T>(DataFrameColumn left, T value)
+        {
+            PrimitiveDataFrameColumn<bool> ret = new PrimitiveDataFrameColumn<bool>(left.Name, left.Length);
             string valString = value.ToString();
-            for (long i = 0; i < Length; i++)
+            for (long i = 0; i < left.Length; i++)
             {
-                ret[i] = (string)this[i] != valString;
+                ret[i] = (string)left[i] != valString;
```

Comment:
For "Arrow strings", wouldn't it be faster to convert `T value` to UTF8, and then do the comparison against the UTF8 values? That way we wouldn't need to materialize all the values as UTF16 strings.